packages/l10n/README.md

```diff
@@ -97,3 +97,271 @@ First, download the data (`core.zip`) for a specific version from the [downloads
+);
+
+console.log(transformedProduct);
+// { name: { en: 'Milk' }, description: { en:  'This is milk' } }
+```
+
+#### When to use it
+
+This transformation tool will be helpful when you consume the commercetools platform `/graphql` API in conjunction with authoring views consuming `@commercetools-frontend/ui-kit`.
+
+Given that you consume:
+
+- The commercetools platform `/graphql` API
+- [`LocalizedTextInput`](https://github.com/commercetools/ui-kit/blob/master/packages/components/inputs/localized-text-input/src/localized-text-input.js)
+
+This will be helpful transforming data from `response -> view`.
+
+```js
+// fetching product from the commercetools platform `/graphql` API
+// returns a product with a `nameAllLocales` and `descriptionAllLocales`
+const product = useMcQuery(ProductQuery, {
+  context: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
+});
+
+// Given that LocalizedTextInput accepts a value of `{ [key: string]: string }`,
+// we tranform our product to match the required shape
+const transformedProduct = applyTransformedLocalizedFields(
+  product,
+  fieldNameTransformMappings
+);
+
+// Finally, we are ready to render our form with the correctly shaped `name` and `description`
+return (
+  <>
+    <LocalizedTextInput name="name" value={transformedProduct.name} />
+    <LocalizedTextInput
+      name="description"
+      value={transformedProduct.description}
+    />
+  </>
+);
+```
+
+### `localize` (DRAFT)
+
+> Transforms a `LocalizedString` to a String
+
+The `localize` util is a util we use internally inside the Merchant Center.
+This util is at the moment subject to change, hence marked as Draft. Use with caution in your Custom Application.
+
+#### Context
+
+As discussed under `applyTransformedLocalizedFields`, a `LocalizedString` is a value type reserved for a `Resource`, e.g a `Product`.
+
+```js
+const product = {
+  name: {
+    en: 'Milk',
+    de: 'Milsch,
+  }
+}
+```
+
+When rendering a Product in a view, we would like to render the value of `name` given the `language` of the UI.
+The `language` is a value controlled by the Merchant Center User (MC User) by updating the language switcher on the Application Bar of the Merchant Center.
+
+The language switcher emits a value that is intended to help us Application Developers display the Resource data (`Product` in our example) in the specified language. Given that the `product.name` has a value of the specified `language`, we can easily pick the value from the `product.name`.
+
+However there are scenarios where the MC User updates the language switcher to a value that our Product example has no matching value. What then? This is where `localize` plays its role.
+
+The `localize` util is authored with a couple of things in mind:
+
+1. Help us Application Developers remain resilient as we attempt to derive a value of `LocalizedString` given a specified language.
+2. Help us Application Developers remain consistent when rendering a value derived from `LocalizedString`
```

Comment:
```suggestion
When rendering a Product in a view, we would like to render the value of a localized string field, such as `name`, given the selected **data locale** of the UI.
The **data locale** is a value controlled by the Merchant Center User (MC User) by changing the **data locale switcher** in the top bar of the Merchant Center. The list of available options is derived by the list of languages specified in the project. The selected value can be read from the **application context**.

However, there might be a case where the selected **data locale** does not match any of the localized string values. In this case, it would be helpful to display a "fallback" value using the `localize` function.

The `localize` util is authored with a couple of things in mind:

1. Help us Custom Application developers remain resilient as we attempt to derive a value of `LocalizedString` given a specified locale.
2. Help us Custom Application developers remain consistent when rendering a value derived from `LocalizedString`.
```